Library/Homebrew/cleaner.rb

```diff
@@ -88,10 +88,19 @@ def clean_dir(d)
 
       Find.prune if @f.skip_clean? path
 
-      next if path.symlink? || path.directory?
+      if path.symlink?
+        if path.extname != ".la" || path.readlink.extname != ".la"
+          next
+        end
+      end
+      next if path.directory?
 
       if path.extname == ".la"
         path.unlink
+      elsif path.symlink?
+        if path.extname == ".la" || path.readlink.extname == ".la"
```

Comment:
The first part will already be handled by `path.extname == ".la"` above. For the latter part please add a check to make sure it's `path.symlink? && path.exist?` as otherwise `path.readlink` will fail and throw an exception.